bolt-sidecar/src/primitives/constraint.rs

```diff
@@ -58,7 +63,12 @@ impl ConstraintsMessage {
         let constraints =
             request.txs.into_iter().map(|tx| Constraint::from_transaction(tx, None)).collect();
 
-        Self { validator_index, slot: request.slot, constraints }
+        Self { validator_index, slot: request.slot, top: false, constraints }
+    }
+
+    /// Returns the total number of leaves in the tree.
+    fn total_leaves(&self) -> usize {
+        4 + self.constraints.len()
```

Comment:
why do we start at 4?

Reply:
Not sure, I took it from the bolt-boost impl by @mempirate

Reply:
My bad I haven't noticed in his PR, but I'm pretty sure it is because the way you encode the struct (which has $$4$$ properties, along with a list) in a binary merkle tree.
@mempirate could you please confirm and document the number in the `bolt-boost` impl as well?

Reply:
Also I'm not sure about this computation since it should take into account chunking and padding: https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md#merkleization

Reply:
We have 3 "unit" types + a list, which will add another leaf for the length + leaves for each individual elements! Will document

Reply:
I realized I haven't tested this - @namn-grg could you add a quick test that generates a random `ConstraintsMessage` and calculates the hash tree root?

Reply:
@mempirate added lmk how it looks?

Reply:
LGTM, we can improve it with the use of `arbitrary` and `arbtest` for fuzzing but random is good enough for now.

Reply:
This test only asserts that `hash_root` is 32 bytes long but that's trivial due to the signature of the `hash_tree_root` method. @mempirate how do we intend to test this? 
In my opinion we should compare the output of this implementation with what fastssz outputs once having generated the hash tree root implementation for this struct using codegen: https://github.com/ferranbt/fastssz/blob/main/sszgen/main.go